### DISCUSSION TITLE
Full name for playlists and scrolling

### DISCUSSION THREAD

**Discussion by Bosturk** [nimsandu/spicetify-bloom#142] [Ideas]
Hi there

First of all, I love the look of this theme, I think especially the "currently playing bar" at the bottom is spot on.

My only gripe, making it for me unusable, is the playlist view. I have very many playlists in many different folders, and I really need their full name to be shown in order to navigate them.
I think for this theme, it would be great if there was a toggle somehow between icons and full text.

Interested to hear your opinions also.

**Comment by nimsandu** [comment]
@Bosturk first of all thanks for reaching out 

Your suggestion is really worthy.
I was thinking to implement something to make it easier.
And i came up with an idea to add a feature which will expand the sidebar by clicking on it.
I've done sketching a ui which might fit.
But it is going to be really challenging to implement this without braking any feature which is available at the movement.
Personally it would take longer than I could imagine to make this work due to my busy studies.
So im hopefully looking forward for every bloom enjoyer to share their opinion to propose a better way to make it easier to navigate through playlists.
Also I would really appreciate if someone volunteer to implement this feature to the theme.

Thanks a lot for sacrificing your time to make bloom, a better theme for every bloom family member.
Regards,
Nimsandu Kottage

**Comment by nimsandu** [comment]
![demo](https://user-images.githubusercontent.com/57834844/209786944-2da10b9a-4e03-466e-a1e3-bc6ac76779bd.jpg)
This is the solution I came up with
this ui will appear once we click the side bar (not on hover because sometimes it is confusing for users).
also, we can add ascending animations and match this functionality with rest of the theme elements without breaking fluent design principles.
looking forward for more recommendations and implementation helps.
Thank You,
enjoy Bloom.

**Reply by nimsandu** [reply]
> ![demo](https://user-images.githubusercontent.com/57834844/209786944-2da10b9a-4e03-466e-a1e3-bc6ac76779bd.jpg) This is the solution I came up with this ui will appear once we click the side bar (not on hover because sometimes it is confusing for users). also, we can add ascending animations and match this functionality with rest of the theme elements without breaking fluent design principles. looking forward for more recommendations and implementation helps. Thank You, enjoy Bloom.

***please note that above ui is only a photoshop illustration and haven't coded yet***

**Comment by SunsetTechuila** [comment]
Just started working and it is so difficult! I really have no idea how to apply and animate all of this in one moment. Maybe someone has examples of similar things that have already been implemented?
<details>
  <summary>Screenshot</summary>
<img src="https://user-images.githubusercontent.com/115353812/230619990-f55cbef6-450f-43c5-9d4a-6c6aa277a44b.png" width="40%" height="40%">
</details>

**Reply by nimsandu** [reply]
you are the first one who started implementing this. take your time, animations can be fixed afterwards!
anyone who is willing to help, you are welcome to work with @SunsetTechuila

**Comment by SunsetTechuila** [comment]
It turns out that Library X provides almost similar functionality. So I will tune it instead of making analogue from scratch

**Reply by SunsetTechuila** [reply]
Unfortunately, there won't be a nice overlay of the panel on the content and background blur, but in life you always have to sacrifice something, don't you?

**Reply by nimsandu** [reply]
@SunsetTechuila yah 🤣. we can improve the look and feel after we find a way to serve the functionality.
